packages/frontend/src/pages/Contribute.jsx

```diff
@@ -264,15 +304,63 @@ export default observer(() => {
       : []
 
     return [
-      'I just contributed to the UniRep trusted setup ceremony!',
+      `I, as ${ceremony.contributionName}, just contributed to the UniRep trusted setup ceremony!`,
       'My circuit hashes are as follows:',
       ...circuitKeys,
     ]
   }
 
+  const postOnGithub = async () => {
+    const access_token = localStorage.getItem('github_access_token')
+    if (!access_token) {
+      await ceremony.oauth('/oauth/github', false, true)
+    } else {
+      const url = await ceremony.postGist()
+      console.log('gist url:', url)
+      setPostMessage({ platform: 'gist', url })
+    }
+  }
+
+  const postOnTwitter = async () => {
+    const url = new URL('/oauth/twitter', HTTP_SERVER)
+    url.searchParams.set('token', ceremony.authToken)
+    const currentUrl = new URL(window.location.href)
+    const dest = new URL('/contribute', currentUrl.origin)
+    url.searchParams.set('redirectDestination', dest.toString())
+    url.searchParams.set(
+      'content',
+      `✨ UniRep Ceremony ✨\nI just contributed to UniRep trusted setup ceremony.\nContribute to help secure the UniRep protocol here: ${currentUrl.origin}\n\nGenerate your own verse and help secure the UniRep protocol.`
+    )
```

Comment:
remove the last line here:
'Generate your own verse and help secure the UniRep protocol'

--and move that text to replace what is currently attached to the link in `index.html`